src/compiler/emitter.ts

```diff
@@ -7735,6 +7342,10 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
                 emitLeadingComments(node.endOfFileToken);
             }
 
+            function emit(node: Node): void {

```

Comment:
I see emitWithoutSourceMap as an explicit statement that the node should not be tied back to its original source location. Since emitStart and emitEnd are no-ops when source mapping is disabled, they will be inlined as a no-op by V8. We likely don't have many cases where we turn source maps on and off during the same session in node.